### PR TITLE
Qcforward check perf zonelog

### PR DESCRIPTION
**Pull request by jcrivenaes**
Making the first jobs in QCForward; check on zone log in wells vs grid

### CODE REVIEW

File: src/fmu/tools/qcforward/qcforward.py [hunk 1]
```diff
@@ -1,12 +1,119 @@
...
-        self._name = None
-        self._whatever = None
+        self._method = None
+        self._data = None
+
+        self._verbosity = 0
+        self._project = None  # Roxar API project
+        self._path = "."
+
+        self._grid = None  # the primary XTGeo Grid() object
```
Comment (tnatt) [inline src/fmu/tools/qcforward/qcforward.py:38]: These variables seems very specific while the QCForward class is quite generic.
E.g. many qc functions will not require a grid.
Perhaps these variables should be set elsewhere?

File: src/fmu/tools/qcforward/qcforward.py [hunk 2]
```diff
@@ -1,12 +1,119 @@
...
+                with open(data, "r") as stream:
+                    xdata = yaml.safe_load(stream)
+            except FileNotFoundError as err:
+                raise RuntimeError(err)
+        else:
+            xdata = data.copy()
+
+        if "dump_yaml" in xdata and xdata["dump_yaml"]:
+            xdata.pop("dump_yaml", None)
+            with open(join(self._path, data["dump_yaml"]), "w") as stream:
```
Comment (tnatt) [inline src/fmu/tools/qcforward/qcforward.py:91]: Consider replacing use of os.path with pathlib.
https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/

```python
open(pathlib.Path(self._path) / data["dump_yaml")
```

File: tests/test_qcforward.py [hunk 3]
```diff
@@ -2,25 +2,83 @@
...
 
+    # now read the dump file:
+    wellcheck.wellzonation_vs_grid("somefile.yml")
 
-def test_zonelog_vs_grid_asrms():
-    """Testing the zonelog vs grid functionality inside RMS"""
+    dfr = pd.read_csv(REPORT)
+    print(dfr)
+    assert dfr.loc[11, "MATCH"] == pytest.approx(58.15, 0.01)
+    os.unlink("somefile.yml")
```
Comment (tnatt) [inline tests/test_qcforward.py:60]: Path("somefile.yml").unlink()

File: src/fmu/tools/qcforward/_wellzonation_vs_grid.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            status = "OK"
+            if match < wlimit:
+                status = "WARN"
+            if match < slimit:
+                status = "STOP"
+
+            match_all.append(match)
+            well_status.append(status)
+        else:
+            match_all.append(0.0)
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_wellzonation_vs_grid.py:156]: Yes wells with missing logs needs better handling

File: src/fmu/tools/qcforward/_parse_data.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+            self.print_info("Grid is already loaded")
+            reuse_grid = True  # grid is already loaded
+        else:
+            self.print_info("Reading grid...")
+            self.print_debug("GRIDNAME: {}".format(gridname))
+            self._grid = xtgeo.grid_from_roxar(self._project, gridname)
+            self._gridname = gridname
+
+    self.print_debug("Looking for zone...")
+    if "zone" in data.keys():
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_parse_data.py:112]: Yes, agree

File: src/fmu/tools/qcforward/_wellzonation_vs_grid.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+    # parsing data stored is self._xxx (general data like grid)
+    self.print_info("Parsing data...")
+    _parse_data.parse(self, data)
+
+    # parse data that are special for this check
+    self.print_info("Parsing additional data...")
+    wzong = _parse_wzong(data)
+
+    match_all = []
```
[the remark this answers is not in the document]
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/_wellzonation_vs_grid.py:117]: Noted

File: src/fmu/tools/qcforward/qcforward.py (hunk 1 above)
Comment (jcrivenaes) [reply src/fmu/tools/qcforward/qcforward.py:38]: I will consider. Otherwise a self._grid = None will tok take any space

**Pull request merged**